src/backend/metal/src/command.rs

```diff
@@ -751,18 +754,25 @@ impl RawCommandQueue<Backend> for CommandQueue {
             if let Some(ref signal_block) = signal_block {
                 msg_send![command_buffer, addCompletedHandler: signal_block.deref() as *const _];
             }
-            // only append the fence handler to the last buffer
-            if i + 1 == num_buffers {
-                if let Some(ref fence) = fence {
-                    let value_ptr = fence.0.clone();
-                    let fence_block = ConcreteBlock::new(move |_cb: *mut ()| -> () {
-                        *value_ptr.lock().unwrap() = true;
-                    }).copy();
-                    msg_send![command_buffer, addCompletedHandler: fence_block.deref() as *const _];
-                }
+            if !inner.retained_buffers.is_empty() {
```

Comment:
If it's deferred, doesn't this mean we will drop `retained_buffers` on first usage (i.e. application may not keep calling `update_buffer`)?

Reply:
Great eye! you spotted a bug :) lemme address it...